fvtest/porttest/omrvmemTest.cpp

```diff
@@ -388,6 +388,122 @@ TEST(PortVmemTest, vmem_test1)
 	reportTestExit(OMRPORTLIB, testName);
 }
 
+/**
+ * Get all the page sizes and make sure we can allocate a memory chunk for each page size.
+ * This particular test is testing the flag OMRPORT_VMEM_MEMORY_MODE_FILE_HANDLE where it
+ * forces default_pageSize_reserve_memory to mmap using a file handle.
+ * 
+ * Checks that each allocation manipulates the memory categories appropriately.
+ */
+TEST(PortVmemTest, vmem_test_file_handle)
+{
+	portTestEnv->changeIndent(1);
+	OMRPORT_ACCESS_FROM_OMRPORT(portTestEnv->getPortLibrary());
+	const char *testName = "omrvmem_test_file_handle";
+	char *memPtr = NULL;
+	uintptr_t *pageSizes = NULL;
+#if defined(J9ZOS390)
+	uintptr_t *pageFlags = NULL;
+#endif /* J9ZOS390 */
+	int i = 0;
+	struct J9PortVmemIdentifier vmemID;
+	char allocName[allocNameSize];
+	int32_t rc = 0;
+	char *lastErrorMessage = NULL;
+	int32_t lastErrorNumber = 0;
+
+	reportTestEntry(OMRPORTLIB, testName);
+
+	/* First get all the supported page sizes */
+	pageSizes = omrvmem_supported_page_sizes();
+#if defined(J9ZOS390)
+	pageFlags =
+#endif /* J9ZOS390 */
+		omrvmem_supported_page_flags();
+
+	/* reserve and commit memory for each page size */
+	for (i = 0 ; pageSizes[i] != 0 ; i++) {
+		uintptr_t initialBlocks = 0;
+		uintptr_t initialBytes = 0;
+
+		/* Sample baseline category data */
+		getPortLibraryMemoryCategoryData(OMRPORTLIB, &initialBlocks, &initialBytes);
+
+		/* reserve and commit */
+#if defined(J9ZOS390)
+		/* On z/OS skip this test for newly added large pages as obsolete omrvmem_reserve_memory() does not support them */
+		if (isNewPageSize(pageSizes[i], pageFlags[i])) {
+			continue;
+		}
+#endif /* J9ZOS390 */
+		memPtr = (char *)omrvmem_reserve_memory(
```

Comment:
What happens when a platform does not support `OMRPORT_VMEM_MEMORY_MODE_FILE_HANDLE`?

Right now this will "work" on all platforms but only Linux (or Unix based?) platforms will actually use a backing file. Can we switch the other platforms to return an unsupported return code? And then modify the test to return success if unsupported is returned? I think that would be cleaner.

@youngar thoughts?

Reply:
I think it's a good idea.  We already have the error code specified `OMRPORT_ERROR_VMEM_NOT_SUPPORTED` in omrporterror.h:218.

Reply:
omrvmem.c:
```
	if(mode & OMRPORT_VMEM_MEMORY_MODE_FILE_HANDLE) {
  		return portLibrary->error_set_last_error(portLibrary,  errno, OMRPORT_ERROR_VMEM_NOT_SUPPORTED);
		return NULL;
	}
```